infra/datalake/processing_stack.py

```diff
@@ -0,0 +1,309 @@
+                        array_size=aws_stepfunctions.JsonPath.number_at("$.content.iteration_size"),
+                        input_path=creation_tasks[task].get("input_path", "$"),
+                        output_path=creation_tasks[task].get("output_path", "$"),
+                        result_path=creation_tasks[task].get("result_path", "$"),
+                        container_overrides=aws_stepfunctions_tasks.BatchContainerOverrides(
+                            command=job_command,
+                            environment=job_environment,
+                        ),
+                        payload=job_payload_parallel,
+                    )
+
+                else:
+                    step_tasks[task] = aws_stepfunctions_tasks.BatchSubmitJob(
+                        self,
+                        f"{task}",
+                        job_name=f"{task}-job",
+                        job_definition=job_definition,
+                        job_queue=batch_job_queue,
+                        input_path=creation_tasks[task].get("input_path", "$"),
+                        output_path=creation_tasks[task].get("output_path", "$"),
+                        result_path=creation_tasks[task].get("result_path", "$"),
+                        container_overrides=aws_stepfunctions_tasks.BatchContainerOverrides(
+                            command=job_command,
+                        ),
+                        payload=job_payload_single,
+                    )
+
+        # success task
+        step_tasks["success"] = aws_stepfunctions.Succeed(
+            self,
+            "success",
+        )
+
+        # STATE MACHINE
+        # state machine definition
+        dataset_version_creation_definition = (
+            step_tasks["check_flat_directory_structure"]
+            .next(step_tasks["content_iterator"])
+            .next(step_tasks["check_files_checksums"])
+            .next(
+                aws_stepfunctions.Choice(self, "content_iteration_finished")
+                .when(
+                    aws_stepfunctions.Condition.not_(
+                        aws_stepfunctions.Condition.number_equals("$.content.next_item", -1)
+                    ),
+                    step_tasks["content_iterator"],
+                )
+                .otherwise(
+                    step_tasks["validation_summary"].next(
+                        aws_stepfunctions.Choice(self, "validation_successful")
+                        .when(
+                            aws_stepfunctions.Condition.boolean_equals(
+                                "$.validation.success", True
+                            ),
+                            step_tasks["success"],
+                        )
+                        .otherwise(step_tasks["validation_failure"])
+                    ),
+                )
+            )
+        )
+
+        # state machine
+        creation_process = aws_stepfunctions.StateMachine(  # pylint:disable=unused-variable
```

Comment:
Why not just don't create the variable, to avoid `pylint` annotations? In the vast majority of cases, whenever Pylint complains I would change the code rather than adding an annotation.

Reply:
Because it will be needed in next stages of development.

Reply:
I'll resolve this, but we should avoid building things we don't yet need (YAGNI) in general. Any work we do which is not connected to *existing* code is a waste until that code is connected to something, and increases the risk of premature design.